vault/vault.go

```diff
@@ -141,13 +141,20 @@ func NewAssumeRoleWithWebIdentityProvider(k keyring.Keyring, config *ProfileConf
 func NewSSORoleCredentialsProvider(k keyring.Keyring, config *ProfileConfig, useSessionCache bool) (aws.CredentialsProvider, error) {
 	cfg := NewAwsConfig(config.SSORegion, config.STSRegionalEndpoints)
 
+	// If we're in an SSH session, we can't use the browser for SSO so we print
+	// the URLs to stdout instead.
+	useStdout := config.SSOUseStdout
+	if os.Getenv("SSH_CONNECTION") != "" {
```

Comment:
this seems like an addition that's independant from the PKCE flow? Overall I wouldn't necessarily expect this "magic" behavior from aws-vault, as a user I prefer a consistent behavior and explicitely passing `--stdout` if needs be

Reply:
Happy to back this out and have it be a separate discussion. My thinking was that given we now have login flows that cater to browser and non-browser setups to direct the user to the one that makes sense in context but I can understand how that would be confusing. A number of us at Tailscale use `aws-vault` with remote workspaces and run into the no-browser scenario frequently.